pkg/generators/rules/omitempty_match_case.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rules
+
+import (
+	"reflect"
+	"strings"
+
+	"k8s.io/gengo/types"
+)
+
+/*
+OmitEmptyMatchCase implements APIRule interface.
+"omitempty" must appear verbatim (no case variants).
+*/
+type OmitEmptyMatchCase struct{}
+
+// Name returns the name of APIRule
+func (n *OmitEmptyMatchCase) Name() string {
+	return "omitempty_match_case"
+}
+
+// Validate evaluates API rule on type t and returns a list of field names in
```

Comment:
That sounds like a generic comment for the `Validate` method, rather than what this method does specifically.

Reply:
whole file was a copy/paste of https://github.com/kubernetes/kube-openapi/blob/master/pkg/generators/rules/names_match.go, can change these if you want